### DISCUSSION TITLE
Add button on Selectable

### DISCUSSION THREAD

**Discussion by Jony01** [ocornut/imgui#4639] [New Users Build/Link/Run issues ONLY!]
How to correctly insert button into combo?

Minimal example contains checkbox showing click action and two example of combos.
```
std::vector<std::string> names({"one", "two", "three"});
static size_t selected_index = 1;
static bool clicked = false;
ImGui::Checkbox("was clicked", &clicked);
ImGui::SameLine();
if (ImGui::Button("del")) clicked = (!clicked);
if (ImGui::BeginCombo("My_combo1", names[selected_index].c_str())) {
    for (const std::string &name : names) { 
        ImGui::PushID(name.c_str());
        size_t index = &name - &names.front();
        bool is_selected = index == selected_index;
        if (ImGui::Selectable(name.c_str(), is_selected)) 
            selected_index = index;            
        ImGui::SameLine();
        ImGui::SetCursorPosX(200);
        if (ImGui::Button("del")) clicked = (!clicked);
        ImGui::PopID(); 
    }
    ImGui::EndCombo();
}
if (ImGui::BeginCombo("My_combo2", names[selected_index].c_str())) {
    for (const std::string &name : names) {
        ImGui::PushID(name.c_str());
        size_t index       = &name - &names.front();
        bool   is_selected = index == selected_index;
        float  pos_x       = ImGui::GetCursorPosX();
        ImGui::SetCursorPosX(200);
        if (ImGui::Button("del")) clicked = (!clicked);
        ImGui::SameLine();
        ImGui::SetCursorPosX(pos_x);
        if (ImGui::Selectable(name.c_str(), is_selected))
            selected_index = index;
        ImGui::PopID();
    }
    ImGui::EndCombo();
}
```
![combo1+2](https://user-images.githubusercontent.com/15888762/136400334-5749b972-a20b-4696-8aea-a4e0636e9a80.gif)

## Combo1
(+) Draw ``Button``s above ``Slectable``
![image](https://user-images.githubusercontent.com/15888762/136403956-4c503e22-cd2c-432b-b709-d386ca5ad3a1.png)
(-)  ``Button`` catch click event only when it is out of ``Selectable``

## Combo2
(+) ``Button``s work correctly (catch click event)
(+) When cursor is above ``Button`` do NOT select line
![image](https://user-images.githubusercontent.com/15888762/136404135-a41ad34b-a232-4b52-b1be-aaa11e8c50a8.png)
(-)  But when hover line it, ``Selectable`` cover button
![image](https://user-images.githubusercontent.com/15888762/136404187-89c4f60b-a044-4923-a52a-4f49448a3086.png)

## Wanted
I would like to combine behave:
- Draw ``Button``s above ``Slectable``
- Buttons catch event on click
- when cursor is above button do NOT select line
- when hover line button is above ``Selectable``

My visually correct but without click response example (similar to "My_combo1" but with using ``Image`` instead of ``Button``):
![wanted](https://user-images.githubusercontent.com/15888762/136398558-68404fbf-f4d5-4581-b943-87d87fb67123.gif)

**Comment by ocornut** [comment]
About hit reaction:
You should use the `ImGuiSelectableFlags_AllowItemOverlap` flag when calling `Selectable()`.

About layout:
Either use AlignTextToFramePadding() before Selectable(), then Button().
Either Selectable() + SmallButton() or a item that match the line height.

**Reply by Jony01** [reply]
Thanks, your tip helps me
